core/controllers/suggestion_test.py

```diff
@@ -322,3 +325,109 @@ def test_suggestion_list_handler(self):
                     suggestion_models.TARGET_TYPE_EXPLORATION, self.EXP_ID)
                 )['suggestions']
             self.assertEqual(len(suggestions), 2)
+
+
+class QuestionSuggestionTests(test_utils.GenericTestBase):
+
+    AUTHOR_EMAIL = 'author@example.com'
+    AUTHOR_EMAIL_2 = 'author2@example.com'
+
+    # Needs to be 12 characteds long.
+    SKILL_ID = 'skill1234567'
+
+    SKILL_DESCRIPTION = 'skill to link question to'
+    TOPIC_ID = 'topic'
+
+    def setUp(self):
+        super(QuestionSuggestionTests, self).setUp()
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+        self.signup(self.AUTHOR_EMAIL, 'author')
+        self.admin_id = self.get_user_id_from_email(self.ADMIN_EMAIL)
+        self.author_id = self.get_user_id_from_email(self.AUTHOR_EMAIL)
+        self.set_admins([self.ADMIN_USERNAME])
+        self.save_new_skill(
+            self.SKILL_ID, self.admin_id, self.SKILL_DESCRIPTION)
+        self.question_dict = {
+            'question_state_data': self._create_valid_question_data(
+                'default_state').to_dict(),
+            'language_code': 'en',
+            'question_state_schema_version': (
+                feconf.CURRENT_EXPLORATION_STATES_SCHEMA_VERSION)
+        }
+        with self.swap(constants, 'USE_NEW_SUGGESTION_FRAMEWORK', True):
+            with self.swap(feconf, 'ENABLE_GENERALIZED_FEEDBACK_THREADS', True):
+                self.login(self.AUTHOR_EMAIL)
+                response = self.testapp.get(feconf.CREATOR_DASHBOARD_URL)
+                csrf_token = self.get_csrf_token_from_response(response)
+
+                self.post_json(
+                    '%s/' % feconf.GENERAL_SUGGESTION_URL_PREFIX, {
+                        'suggestion_type': (
+                            suggestion_models.SUGGESTION_TYPE_ADD_QUESTION),
+                        'target_type': suggestion_models.TARGET_TYPE_TOPIC,
+                        'target_id': self.TOPIC_ID,
+                        'target_version_at_submission': 1,
+                        'change_cmd': {
+                            'cmd': (
+                                question_domain
+                                .CMD_ADD_QUESTION_FROM_SUGGESTION),
+                            'question_dict': self.question_dict,
+                            'skill_id': self.SKILL_ID
+                        },
+                        'description': 'Add new question to skill'
+                    }, csrf_token=csrf_token)
+                self.logout()
+
+    def test_query_question_suggestions(self):
+        with self.swap(constants, 'USE_NEW_SUGGESTION_FRAMEWORK', True):
+            suggestions = self.get_json(
+                '%s?suggestion_type=%s' % (
+                    feconf.GENERAL_SUGGESTION_LIST_URL_PREFIX,
+                    suggestion_models.SUGGESTION_TYPE_ADD_QUESTION)
+                )['suggestions']
+            self.assertEqual(len(suggestions), 1)
```

Comment:
Should probably check the details of the retrieved suggestion, too.

Reply:
Done